selftests/functional/test_output.py

```diff
@@ -161,6 +161,50 @@ def test3(self):
         sys.exit(j.run())
 """
 
+AVOID_NON_TEST_TASKS = """#!/usr/bin/env python3
+
+from avocado.core import nrunner
+from avocado.core.job import Job
+from avocado.core.task.runtime import RuntimeTask
+from avocado.core.test_id import TestID
+from avocado.plugins import runner_nrunner
+from avocado.utils.network.ports import find_free_port
+
+
+class RunnerNRunnerWithFixedTasks(runner_nrunner.Runner):
+    @staticmethod
+    def _get_all_runtime_tasks(test_suite):
+        runtime_tasks = []
+        no_digits = len(str(len(test_suite)))
+        status_uris = [test_suite.config.get('nrunner.status_server_uri')]
+        for index, runnable in enumerate(test_suite.tests, start=1):
+            prefix = index
+            test_id = TestID(prefix, runnable.uri, None, no_digits)
+            if index % 2 == 1:
```

Comment:
I find this a bit cryptic... IMO it's hard to read and can have unwanted consequences if we say add to the list of tests.  Wouldn't it be better to check say the `uri` and always create a `non-test` task when it's not `/bin/true`?